trace.go

```diff
@@ -83,11 +83,20 @@ type payload struct {
 	// TODO(dsymonds): add stringifying info to codec, and limit how much we hold here?
 }
 
+const truncateSize = 100
+
+func truncate(x string, l int) string {
+	if l > len(x) {
+		return x
+	}
+	return x[:l]
```

Comment:
This truncation is incorrect if the string contains UTF-8 multibyte characters. It can produce invalid UTF-8 sequences.

Reply:
This is true, but is that a problem?